components/tools/OmeroPy/src/omero/tables.py

```diff
@@ -229,14 +229,32 @@ def openfile(self, mode):
                             self.__hdf_path, mode))
                     mode = "r"
 
-            return tables.openFile(str(self.__hdf_path), mode=mode,
-                                   title="OMERO HDF Measurement Storage",
-                                   rootUEP="/")
+            return tables.open_file(str(self.__hdf_path), mode=mode,
+                                    title="OMERO HDF Measurement Storage",
+                                    rootUEP="/")
         except (tables.HDF5ExtError, IOError) as e:
             msg = "HDFStorage initialized with bad path: %s: %s" % (
                 self.__hdf_path, e)
             self.logger.error(msg)
             raise omero.ValidationException(None, None, msg)
+        except (ValueError) as e:
+
+            # trap PyTables >= 3.1 FILE_OPEN_POLICY exception
+            # to provide an updated message
+            if 'FILE_OPEN_POLICY' in str(e):
+                e = ValueError(
+                    "PyTables [{version}] no longer supports opening multiple "
+                    "files\n"
+                    "even in read-only mode on this HDF5 version "
+                    "[{hdf_version}]. You can accept this\n"
+                    "and not open the same file multiple times at once,\n"
+                    "upgrade the HDF5 version, or downgrade to PyTables 3.0.0 "
+                    "which allows\n"
+                    "files to be opened multiple times at once\n"
+                    .format(version=tables.__version__,
+                            hdf_version=tables.get_hdf5_version()))
```

Comment:
In previous versions of pytables, hdflist prevents you from opening the file but that was not prevented at the level of pytables when using openFile
``addOrThrow`` will always throw at open_file w/o strict and ignore 
```
fileno = hdffile.fileno()

        try:
            portalocker.lockno(
                fileno, portalocker.LOCK_NB | portalocker.LOCK_EX)
        except portalocker.LockException:
            hdffile.close()
            raise omero.LockTimeout(
                None, None,
                "Cannot acquire exclusive lock on: %s" % hdfpath, 0)
        except:
            hdffile.close()
            raise
```